third_party/amd/lib/TritonAMDGPUTransforms/RescheduleOps.cpp

```diff
@@ -1406,60 +1634,280 @@ SchedDagNode *getOriginalOrder(SchedDagNode *a, SchedDagNode *b) {
   return (a->id < b->id) == (Direction == SchedDirection::TopDown) ? a : b;
 }
 
+/******************************************************************************
+  Each PriorityCalculator gets to see the current dag
+  and set scheduling priorities to nodes.
+******************************************************************************/
+struct PriorityCalculator {
+  PriorityCalculator(SchedDagNodePriorityType type) : priorityType(type) {}
+
+  virtual void calcPriorities() = 0;
+
+  void addPrioritiesToDag(SchedDag *d) {
+    LDBG("PriorityCalculator<" << toString(priorityType)
+                               << ">::addPrioritiesToDag()");
+    dag = d;
+    calcPriorities();
+  }
```

Comment:
I would be more explicit here regarding `SchedDag *d`. I would pass the pointer to a dag to `calcPriorities` and remove `SchedDag *dag;` data member of the `PriorityCalculator` class to avoid confusion. 

Note: if it is still needed then, in theory, `SchedDag *dag;` must be initialized in the constructor.